phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4125,7 +4125,7 @@ public MutationState createSchema(CreateSchemaStatement create) throws SQLExcept
 
     private void validateSchema(String schemaName) throws SQLException {
         if (SchemaUtil.NOT_ALLOWED_SCHEMA_LIST.contains(
-                schemaName.toUpperCase())) { throw new SQLExceptionInfo.Builder(SQLExceptionCode.SCHEMA_NOT_ALLOWED)
```

Comment:
This code would automatically uppercase the schema irrespective of considering if it's enclosed in quotes or not. It will use that value to validate the schema. However, will creating, it will use the regular name. Checkout Line 4093. This seems an inconsistent logic to me. 
For the query
`CREATE SCHEMA "ABC"`, this code will create to create a schema with name `"ABC"` with the double quotes. There was no test that was actually testing this and thus it wasn't caught yet. I wrote a test in `QueryParserTest` and caught it.
@JamesRTaylor Please advice.

Reply:
Ah, I see. Good catch. Your change is the right one.

Reply:
Also checkout the test I updated in `CreateSchemaIT` class.

Reply:
@karanmehta93 , I tried case-sensitive schemas with master code, I don't see any problem. can you check if normalization is happening at some other(or not at the right) place?
```
0: jdbc:phoenix:localhost> create schema "ABC";
No rows affected (0.224 seconds)
```
```
0: jdbc:phoenix:localhost> select TABLE_SCHEM from system.catalog;
+--------------+
| TABLE_SCHEM  |
+--------------+
| ABC          |
```
```
hbase(main):002:0> list_namespace
NAMESPACE
ABC
```
```
0: jdbc:phoenix:localhost> create schema "small_case"
. . . . . . . . . . . . .> ;
No rows affected (0.424 seconds)
0: jdbc:phoenix:localhost> select TABLE_SCHEM from system.catalog;
+--------------+
| TABLE_SCHEM  |
+--------------+
| small_case   |

hbase(main):003:0> list_namespace
NAMESPACE
small_case
```

Reply:
@ankitsinghal 
You are correct, the schema gets normalized when we try to create a `PSchema` object and we use `PSchema.getSchemaName()` method which returns the correct schema name. Thanks for pointing it out.
However, the issue is that the normalization doesn't happen before checking the validity of the schema. As of now, the schema is uppercased even before checking if its without quotes or with quotes. This leads to issue when you try to create schema with the names DEFAULT or HBASE, which is the focus for this JIRA.
I will update the patch accordingly.

Reply:
@ankitsinghal Thanks for your comments. Please checkout v2 of the patch. I have fixed the bug and I updated the test to verify those scenarios. I also made sure that we don't normalize schema names multiple times, else schema names will always be in capital letters.
FYI. @twdsilva